app/models/pipeline_run_stage.rb

```diff
@@ -123,26 +127,21 @@ def due_for_aegea_check?
   end
 
   def update_job_status
-    if !id || !started? || succeeded? || failed?
-      # This is called only from PipelineRun.update_status and only when the guard above is false.
+    if !id || !started?
```

Comment:
It appears the changes to this function are not all here yet.   I look forward to reviewing the more complete version.

Reply:
Actually they're all there :) Just, github was probably hiding part of the diff from you because it was to large (pipeline_run.rb)